src/storage/src/hummock/compactor/compactor_runner.rs

```diff
@@ -191,21 +192,29 @@ impl CompactorRunner {
                     task_progress.clone(),
                     compact_io_retry_time,
                 ));
+            } else if tables.len() > MAX_OVERLAPPING_SST {
+                let sst_groups = partition_overlapping_sstable_infos(tables);
```

Comment:
minor: Inspired by @hzxa21 , TierCompaction allows the selection of consecutive overlapping and non-overlapping levels. Therefore, we can directly filter out the non-overlapping level to reduce the calculation